dotCMS/src/main/java/com/dotcms/storage/FileSystemStoragePersistenceAPIImpl.java

```diff
@@ -0,0 +1,461 @@
+                    ", is null, does not exist can not be read or bucket: " + groupName +
+                    " could not be written");
+        }
+
+        return true;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param groupName {@link String} the group to upload
+     * @param path       {@link String} path to upload the file
+     * @param writerDelegate     {@link ObjectWriterDelegate} stream to upload
+     * @param object     {@link Serializable} object to write into the storage
+     * @param extraMeta  {@link Map} optional metadata, this could be null but depending on the implementation it would need some meta info.
+     * @return
+     */
+    @Override
+    public Object pushObject(final String groupName, final String path,
+            final ObjectWriterDelegate writerDelegate,
+            final Serializable object, final Map<String, Object> extraMeta) {
+
+        if (!this.existsGroup(groupName)) {
+
+            throw new IllegalArgumentException("The bucketName: " + groupName +
+                    ", does not have any file mapped");
+        }
+
+        final File groupFile = this.groups.get(groupName.toLowerCase());
+
+        if (groupFile.canWrite()) {
+
+            try {
+
+                final File destBucketFile = new File(groupFile, path.toLowerCase());
+                final String compressor = Config
+                        .getStringProperty("CONTENT_METADATA_COMPRESSOR", "none");
+                this.prepareParent(destBucketFile);
+
+                try (OutputStream outputStream = FileUtil
+                        .createOutputStream(destBucketFile.toPath(), compressor)) {
+
+                    writerDelegate.write(outputStream, object);
+                    outputStream.flush();
+                }
+            } catch (IOException e) {
+
+                Logger.error(this, e.getMessage(), e);
+                throw new DotRuntimeException(e);
+            }
+        } else {
+
+            throw new IllegalArgumentException("The bucket: " + groupName + " could not write");
+        }
+
+        return true;
+    }
+
+    /**
+     * makes parent dir if doesnt exist
+     * @param file
+     */
+    private void prepareParent(final File file) {
+
+        if (!file.getParentFile().exists()) {
+
+            file.getParentFile().mkdirs();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param groupName {@link String} the bucket to push
+     * @param path       {@link String} path to push the file
+     * @param file       {@link File}   the actual file
+     * @param extraMeta  {@link Map} optional metadata, this could be null but depending on the implementation it would need some meta info.
+     * @return
+     */
+    @Override
+    public Future<Object> pushFileAsync(final String groupName, final String path,
+            final File file, final Map<String, Object> extraMeta) {
+        return DotConcurrentFactory.getInstance().getSubmitter("StoragePool").submit(
```

Comment:
Codacy found an issue: [The String literal "StoragePool" appears 4 times in this file; the first occurrence is on line 210](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)